src/generator.js

```diff
@@ -72,17 +72,21 @@ class Generator {
   heading2ExplicitId = new Map();
 
   /**
-   * @param {string} lang
-   * @param {string} version
-   * @param {string} srcDir
-   * @param {string} outDir
-   * @param {GeneratorFormatter} formatter
+   * @param {{
+   * lang: string,
+   * version: string,
+   * srcDir: string,
+   * outDir: string,
+   * formatter: GeneratorFormatter,
+   * renderSyncNoArgsMethodAsProperty?: boolean,
```

Comment:
nit: I think this is more of a `GeneratorFormatter` property rather than generator's parameter, but this way also works.